src/js/components/Diagram/index.d.ts

```diff
@@ -4,10 +4,22 @@ import { ColorType } from '../../utils';
 
 export type DiagramConnectionAnchor = 'center' | 'vertical' | 'horizontal';
 export type DiagramConnectionType = 'direct' | 'curved' | 'rectilinear';
+export type DiagramAnimationType =
+  | boolean
```

Comment:
Did you try these out in a .tsx storybook example to make sure all the typings are working as expected? asking from past experience of it not working as I expect hahaa

Reply:
The .tsx migration went well, gave the animation's arguments unexpected results and everything functions as expected.